java/vector/src/main/codegen/templates/VariableLengthVectors.java

```diff
@@ -370,13 +370,20 @@ public void reset() {
   }
 
   public void reAlloc() {
-    final long newAllocationSize = allocationSizeInBytes*2L;
+    long baseSize = allocationSizeInBytes;
+    final int currentBufferCapacity = data.capacity();
+    if (baseSize < (long)currentBufferCapacity) {
```

Comment:
Seems to be same as `FixedValueVector`. Maybe some refactor?

Reply:
Sorry, I don't understand the comment. The same problem is fixed across reAlloc() of all vectors --  bit vector, fixed width and variable width.

Reply:
Oh, I meant, since these logic looks similar, I am wondering if we can refactor the shared logic into the base class.

Reply:
I had thought about it but since with ARROW-1463 the inheritance hierarchy and templates might look different, I thought may there is not much gain in refactoring now?

Reply:
I see. Yeah I don't love it but if we are going to fix this in ARROW-1463 then I am ok.